### DISCUSSION TITLE
Problem with API

### DISCUSSION THREAD

**Discussion by kruegge82** [kimai/kimai#4305] [Questions & Answers] [answered]
Hello,

since a view days a i had  problems with kimai, @kevinpapst helped with the solution adding the sql charset and Serverversion.
first i think all is working but, no the API is not working...

Service Unavailable
The server is temporarily unable to service your request due to maintenance downtime or capacity problems. Please try again later.

Example request:

```
    $ch = curl_init();

    curl_setopt($ch, CURLOPT_URL, 'https://domain.com/api/users?visible=1&term=39);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
    curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'GET');

    $headers = array();
    $headers[] = 'Accept: application/json';
    $headers[] = 'X-Auth-User: user';
    $headers[] = 'X-Auth-Token: pass';
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

    $result = curl_exec($ch);
    if (curl_errno($ch)) {
        echo 'Error:' . curl_error($ch);
    }
    curl_close($ch);
    $result= json_decode($result);

```
the value "39" is stored in one user as Personal Number (Personalnummer)
In the Log are no errors only infos.

If i chnage to dev mode all is working.
Version 2.0.34
last changes: add "?charset=utf8mb4&serverVersion=10.1.48-MariaDB" to the DATABSE_URL
This is the only serverVersion which is working, my server tells me: 10.6.12-MariaDB-0ubuntu0.22.04.1-log

I hope anyone have a solution...

**Comment by kevinpapst** [comment]
Check your server logs if there are no errors in Kimai logs.
Make sure that Kimai can actually write the log! Doctor screen will help.
If it works in dev and not in prod => it is likely a caching / file permission issue.
run `rm -r var/cache/*` in your kimai directory and then fix file permissions as explained in the docs

**Comment by kruegge82** [comment]
Hello,

i have now updated the permissions and delete the cache folder...
and also i get no new log entries...
but i have on bad behavior...
the first api call is working, the secon is not working.
After delete cache folder, first call is working following calls are not working.

**Reply by kevinpapst** [reply, marked as answer]
🤷  not a Kimai bug, but an environment issue. 
File permissions or shared host with broken PHP extension like
https://github.com/kimai/kimai/discussions?discussions_q=is%3Aopen+ioncube

**Reply by kruegge82** [reply]
ioncube.loader.encoded_paths = /www/htdocs/w00000aa/logs

this line helps... its working, thank you
